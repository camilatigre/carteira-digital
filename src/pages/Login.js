import React, { useState } from 'react';
import '../styles/common/Forms.css';
import Header from '../common/Header';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import { Link, BrowserRouter, useHistory } from 'react-router-dom';
import { fieldsValidation } from '../utils/validation';

const Login = () => {
  let history = useHistory();

  const [email, setEmail] = useState({ name: 'email', value: '', helper: '' });
  const [password, setPassword] = useState({
    name: 'password',
    value: '',
    helper: '',
  });

  const onChange = (evt) => {
    const { name, value } = evt.target;

    switch (name) {
      case 'email':
        setEmail({ name, value, helper: '' });
        break;
      case 'password':
        setPassword({ name, value, helper: '' });
        break;
      default:
        return;
    }
  };

  const enterOnApp = () => {
    const notValidFields = fieldsValidation([email, password]);

    if (notValidFields.length > 0) {
      notValidFields.forEach((field) => {
        if (field.name === 'email') {
          setEmail(field);
        }

        if (field.name === 'password') {
          setPassword(field);
        }
      });
      return;
    }

    const hasSomeoneLoggedIn = localStorage.getItem('login') !== '';

    if (hasSomeoneLoggedIn) {
      localStorage.clear();
    }

    localStorage.setItem('login', email.value);
    history.push('/home');
  };

  return (
    <div className="outsideBody">
      <Header whiteLogo={true} />
      <div id="form">
        <div className="container">
          <p className="paragraphGreen" align="center">
            Faça aqui seu login para começar suas transações!
          </p>
          <TextField
            className="textField"
            fullWidth
            label="E-MAIL"
            type="email"
            name="email"
            required
            variant="outlined"
            onChange={onChange}
            helperText={email['helper']}
            value={email['value']}
            placeholder={'Digite seu E-MAIL'}
            inputProps={{
              id: 'idJestEmail',
            }}
          />
          <TextField
            className="textField"
            fullWidth
            label="SENHA"
            type="password"
            name="password"
            required
            variant="outlined"
            onChange={onChange}
            helperText={password['helper']}
            value={password['value']}
            placeholder={'Digite sua SENHA'}
            inputProps={{
              id: 'idJestPassword',
            }}
          />
          <div className="buttonContainer">
            <BrowserRouter>
              <Link to="/register">
                <Button variant="outlined" className="secondaryButton">
                  Quero me cadastrar
                </Button>
              </Link>
              <Button id="enterButton" onClick={enterOnApp} variant="contained" className="primaryButton">
                Entrar
              </Button>
            </BrowserRouter>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Login;
