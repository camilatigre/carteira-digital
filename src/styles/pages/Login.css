/* Arquivo de estilos que impactam apenas na p√°gina de Login */
body {
  background-color: #00ab63;
}

#login {
  display: flex;
  align-items: center;
  justify-content: center;
  height: calc(100vh - 100px);
}

#login .container {
  width: 50%;
  height: 300px;
  background-color: #fff;
  padding: 20px;
  border-radius: 10px;
  box-shadow: 0px 0px 15px 3px #018a50;
}

#login .textField {
  margin: 20px 0 0 0;
}
