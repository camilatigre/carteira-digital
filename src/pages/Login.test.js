import React from 'react';
import Login from './Login';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MemoryRouter } from 'react-router-dom';
import { render, fireEvent, screen } from '@testing-library/react';

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe('<Login />', () => {
  it('Deveria renderizar a pagina de login', () => {
    const page = shallow(<Login />);
    expect(toJson(page)).toMatchSnapshot();
  });

  it('trys to loggin when button clicked', () => {
    render(
      <MemoryRouter>
        <Login />
      </MemoryRouter>,
    );

    fireEvent.change(screen.getByPlaceholderText('Digite seu E-MAIL'), { target: { value: 'exemple@gmail.com' } });
    fireEvent.change(screen.getByPlaceholderText('Digite sua SENHA'), { target: { value: '1234' } });
    fireEvent.click(screen.getByText('Entrar'));

    expect(mockHistoryPush).toHaveBeenCalledWith('/home');
  });
});
