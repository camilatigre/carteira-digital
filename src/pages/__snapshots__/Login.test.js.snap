// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> Deveria renderizar a pagina de login 1`] = `
<div
  className="outsideBody"
>
  <Header
    whiteLogo={true}
  />
  <div
    id="form"
  >
    <div
      className="container"
    >
      <p
        align="center"
        className="paragraphGreen"
      >
        Faça aqui seu login para começar suas transações!
      </p>
      <WithStyles(ForwardRef(TextField))
        className="textField"
        fullWidth={true}
        helperText=""
        inputProps={
          Object {
            "id": "idJestEmail",
          }
        }
        label="E-MAIL"
        name="email"
        onChange={[Function]}
        placeholder="Digite seu E-MAIL"
        required={true}
        type="email"
        value=""
        variant="outlined"
      />
      <WithStyles(ForwardRef(TextField))
        className="textField"
        fullWidth={true}
        helperText=""
        inputProps={
          Object {
            "id": "idJestPassword",
          }
        }
        label="SENHA"
        name="password"
        onChange={[Function]}
        placeholder="Digite sua SENHA"
        required={true}
        type="password"
        value=""
        variant="outlined"
      />
      <div
        className="buttonContainer"
      >
        <BrowserRouter>
          <Link
            to="/register"
          >
            <WithStyles(ForwardRef(Button))
              className="secondaryButton"
              variant="outlined"
            >
              Quero me cadastrar
            </WithStyles(ForwardRef(Button))>
          </Link>
          <WithStyles(ForwardRef(Button))
            className="primaryButton"
            id="enterButton"
            onClick={[Function]}
            variant="contained"
          >
            Entrar
          </WithStyles(ForwardRef(Button))>
        </BrowserRouter>
      </div>
    </div>
  </div>
</div>
`;
